R/subset.R

```diff
@@ -30,13 +30,18 @@ str_subset <- function(string, pattern, negate = FALSE) {
   check_lengths(string, pattern)
   check_bool(negate)
 
-  switch(type(pattern),
-         empty = no_empty(),
-         bound = no_boundary(),
-         fixed = string[str_detect(string, pattern, negate = negate)],
-         coll  = string[str_detect(string, pattern, negate = negate)],
-         regex = string[str_detect(string, pattern, negate = negate)])
+  if (type(pattern) == "empty") no_empty()
+  if (type(pattern) == "bound") no_boundary()
 
+  idx <- str_detect(string, pattern, negate = negate)
```

Comment:
I'd suggest keeping the original `switch` as this makes it easier to verify that every case is correctly handled.